### PR TITLE
Improve Pacifist effect to make sense.

### PR DESCRIPTION
**Pull request by ProfessorBiddle**
With the help of Pongo, finally found a way to detect when the player hits someone. This works way better and makes much more sense than the old effect.

### CODE REVIEW

File: ChaosMod/Effects/db/Player/PlayerPacifist.cpp [hunk 1]
```diff
@@ -2,19 +2,43 @@
 //Effect by ProfessorBiddle, requested thousands of times on discord
 static void OnTick()
 {
-	Player player = PLAYER_ID();
+	int hits0, hits1, hits2;
+	STAT_GET_INT(GET_HASH_KEY("SP0_HITS"), &hits0, -1);
+	STAT_GET_INT(GET_HASH_KEY("SP1_HITS"), &hits1, -1);
+	STAT_GET_INT(GET_HASH_KEY("SP2_HITS"), &hits2, -1);
```
Comment (Last0xygen) [inline ChaosMod/Effects/db/Player/PlayerPacifist.cpp:8]: wouldnt it make more sense to make it so:
* you hold the current player hash and current hits 
* Check on Tick if the player changed
  * if so, set the current hits to the current player hits 
* if not, compare the current hits to the last saved hits
  * if > last hits -> kill
  * if <= hits -> set to saved hits (may happen after mission restart / loading a savegame?)
* Also, ONLY get the hash within the switch case, the rest is always the same procedure and just duplicates the code

Just my 2 cents

**Review by Last0xygen — CHANGES_REQUESTED** [review]

File: ChaosMod/Effects/db/Player/PlayerPacifist.cpp (hunk 1 above)
Comment (ProfessorBiddle) [reply ChaosMod/Effects/db/Player/PlayerPacifist.cpp:8]: Thanks for the input, I'll do this.

**Comment by TheRealAyCe** [comment]
Are these stats affected by the characters killing someone while NPC AI takes over?

**Comment by ProfessorBiddle** [comment]
> Are these stats affected by the characters killing someone while NPC AI takes over?

I'm not sure, it should just count hits by whatever character the player is controlling.

File: ChaosMod/Effects/db/Player/PlayerPacifist.cpp [hunk 2]
```diff
@@ -2,17 +2,50 @@
...
+	switch (playerHash)
+	{
+	case 225514697: // Michael 
+		hitHash = GET_HASH_KEY("SP0_HITS");
+		break;
+	case 2602752943: // Franklin
+		hitHash = GET_HASH_KEY("SP0_HITS");
+		break;
+	case 2608926626: // Trevor
+		hitHash = GET_HASH_KEY("SP0_HITS");
```
Comment (pongo1231) [inline ChaosMod/Effects/db/Player/PlayerPacifist.cpp:21]: You are checking for the same stats as above here

File: ChaosMod/Effects/db/Player/PlayerPacifist.cpp [hunk 3]
```diff
@@ -2,17 +2,50 @@
...
+		switch (playerHash)
+		{
+		case 225514697: // Michael 
+			hitHash = GET_HASH_KEY("SP0_HITS");
+			break;
+		case 2602752943: // Franklin
+			hitHash = GET_HASH_KEY("SP0_HITS");
+			break;
+		case 2608926626: // Trevor
+			hitHash = GET_HASH_KEY("SP0_HITS");
```
Comment (pongo1231) [inline ChaosMod/Effects/db/Player/PlayerPacifist.cpp:42]: Same applies here also

**Review by pongo1231 — CHANGES_REQUESTED** [review]

File: ChaosMod/Effects/db/Player/PlayerPacifist.cpp (hunk 2 above)
Comment (ProfessorBiddle) [reply ChaosMod/Effects/db/Player/PlayerPacifist.cpp:21]: Whoops, didn't test with the other two characters 📦 
I'll fix this now.

**Comment by Last0xygen** [comment]
Couldn’t you just keep the player and lastHitcount „outside“ of onTick?
Set it on OnStart to hash -1 and count -1, then you only need one STAT_GET_INT inside the onTick 🤷‍♂️
no WAIT(0) and so on, I hope you understand what I mean 😅

File: ChaosMod/Effects/db/Player/PlayerPacifist.cpp [hunk 4]
```diff
@@ -2,17 +2,50 @@
...
+	}
+	STAT_GET_INT(hitHash, &playerHits, -1);
+
+	WAIT(0);
 
-	if (GET_ENTITY_PLAYER_IS_FREE_AIMING_AT(player, &target))
+	//check if player changed
+	playerPed = PLAYER_PED_ID();
+	Hash newPlayerHash = GET_ENTITY_MODEL(playerPed);
+	if (newPlayerHash = playerHash)
```
Comment (pongo1231) [inline ChaosMod/Effects/db/Player/PlayerPacifist.cpp:31]: `==` instead of `=`

**Comment by ProfessorBiddle** [comment]
> Couldn’t you just keep the player and lastHitcount „outside“ of onTick?
> Set it on OnStart to hash -1 and count -1, then you only need one STAT_GET_INT inside the onTick 🤷‍♂️
> no WAIT(0) and so on, I hope you understand what I mean 😅

I think I do, I'll give it a shot. I'm pretty inexperienced with c++ and programming in general, so thanks for all the help!

**Comment by Last0xygen** [comment]
> 
> 
> > Couldn’t you just keep the player and lastHitcount „outside“ of onTick?
> > Set it on OnStart to hash -1 and count -1, then you only need one STAT_GET_INT inside the onTick 🤷‍♂️
> > no WAIT(0) and so on, I hope you understand what I mean 😅
> 
> I think I do, I'll give it a shot. I'm pretty inexperienced with c++ and programming in general, so thanks for all the help!

if you need any help just pm me on discord ;)

File: ChaosMod/Effects/db/Player/PlayerPacifist.cpp [hunk 5]
```diff
@@ -1,20 +1,42 @@
 #include <stdafx.h>
 //Effect by ProfessorBiddle, requested thousands of times on discord
-static void OnTick()
-{
-	Player player = PLAYER_ID();
 
-	Entity target;
+int playerHits, playerHitsLast;
```
Comment (pongo1231) [inline ChaosMod/Effects/db/Player/PlayerPacifist.cpp:4]: Make this `static` please

**Review by pongo1231 — CHANGES_REQUESTED** [review]

**Pull request merged**